packages/components/src/utils/select/keyboard-behavior.ts

```diff
@@ -0,0 +1,249 @@
+export const SELECT_SEARCH_TIMEOUT: number = 500;
+// The amount of options to be jumped when performing a page-based navigation using PageUp or PageDown.
+const PAGE_UP_DOWN_STEP_AMOUNT: number = 10;
+
+// TODO: Improve this to be a smooth transition
+/**
+ * Handles scrolling within the list to ensure that the highlighted item is always visible.
+ * @param {HTMLElement} scrollElement - The HTML element to be scrolled.
+ * @param {HTMLElement} element - The element to scroll to.
+ * @returns {void}
+ */
+export const handleSelectDropdownScroll = (scrollElement: HTMLElement, element: HTMLElement): void => {
+  const { maxHeight } = getComputedStyle(scrollElement);
+  const hostElementHeight = parseInt(maxHeight, 10);
+  if (scrollElement.scrollHeight > hostElementHeight) {
+    element.scrollIntoView({ block: 'nearest' });
+  }
+};
+
+/**
+ * Determines the action to be taken based on a keyboard event and the state of the select menu.
+ *
+ * @param {KeyboardEvent} event - The keyboard event triggering the action.
+ * @param {boolean} menuOpen - A boolean indicating whether the select menu is open or closed.
+ * @returns {selectActions} - The corresponding action to be performed.
+ */
+export const getActionFromKey = (event: KeyboardEvent, menuOpen: boolean): SelectAction => {
+  const { key, altKey, ctrlKey, metaKey } = event;
+  const openKeys = ['ArrowDown', 'ArrowUp', 'Enter', ' ']; // all keys that will do the default open action
+  // handle opening when closed
+  if (!menuOpen && openKeys.includes(key)) {
+    return selectActions.Open;
+  }
+
+  // home and end move the selected option when open or closed
+  if (key === 'Home') {
+    return selectActions.First;
+  }
+  if (key === 'End') {
+    return selectActions.Last;
+  }
+
+  // handle typing characters when open or closed
+  if (key === 'Backspace' || key === 'Clear' || (key.length === 1 && key !== ' ' && !altKey && !ctrlKey && !metaKey)) {
+    return selectActions.Type;
+  }
+
+  // handle keys when open
+  if (menuOpen) {
+    if (key === 'ArrowUp' && altKey) {
+      return selectActions.CloseSelect;
+    } else if (key === 'ArrowDown' && !altKey) {
+      return selectActions.Next;
+    } else if (key === 'ArrowUp') {
+      return selectActions.Previous;
+    } else if (key === 'PageUp') {
+      return selectActions.PageUp;
+    } else if (key === 'PageDown') {
+      return selectActions.PageDown;
+    } else if (key === 'Escape') {
+      return selectActions.Close;
+    } else if (key === 'Enter' || key === ' ' || key === 'Tab') {
+      return selectActions.CloseSelect;
+    }
+  }
+};
+
+/**
+ * Gets the updated index based on the current index, maximum index, and the select action.
+ *
+ * @param {number} currentIndex - The current index in the list of options.
+ * @param {number} maxIndex - The maximum index in the list of options.
+ * @param {selectActions} action - The select action indicating how to update the index.
```

Comment:
Looks like several jsdocs weren't replaced/refactored.
```suggestion
 * @param {SelectAction} action - The select action indicating how to update the index.
```